src/frontend/dash/circlePacker.ts

```diff
@@ -1,203 +1,351 @@
+    abstract updateSize();
 }
 
-function getDistanceFrom(circle1, x, y) {
-    return Math.sqrt(Math.pow(circle1.x - x, 2) + Math.pow(circle1.y - y, 2));
+export class MemberCircle extends Circle {
+    loginTime : number;
+    name : string;
+
+    constructor(loginTime, name, imgurl) {
+        super(Math.sqrt(.2) * 10);
+
+        this.loginTime = loginTime;
+
+        this.element.style.backgroundImage = `url(${imgurl})`;
+        this.element.className = 'memberCircle'
+
+        const nameBubble = this.element.appendChild(document.createElement('name'));
+        this.name = nameBubble.innerHTML = name
+        nameBubble.className = 'bubblename'
+        nameBubble.style.backgroundColor = BUBBLE_COLORS[Math.floor(Math.random() * BUBBLE_COLORS.length)];
+        // name.style.fontSize = `${Math.min(30*multiplier, 20)}px`;
+        nameBubble.style.fontSize = '10px';
+    }
+
+    updateSize() {
+        this.r = Math.sqrt((Date.
+        now() - this.loginTime) / 36000 + .2) * 10;
+    }
 }
 
-function placeCircle(circle) {
-    if(!placedCircles.length) {
-        circle.x = circle.y = 0;
-        return;
+export class ClockCircle extends Circle{
+    constructor() {
+        super(Math.sqrt(.2) * 10);
+        this.element.className = 'clockCircle'
+        this.element.innerHTML = 
+        `<div class="timestack">
+        <div class="time">
+            <div class="colon">
+                
+            </div>
+            <div class="numbers">
+                <div class="hours" id="hoursCircleText">
+                11:35
+                </div>
+                
+            </div>
+        </div>
+        <div class="businfo">
+        <div class="busstack">
+            <div class="busname">
+            Gresham
+        <img src="../static/img/trimet-logo.png" class="trimetlogo">
+            </div>
+            <div class="bustime east ">--&nbsp;min</div>
+            <div class="bustime smoler east">--&nbsp;min</div>
+
+        </div>
+        <div class="busstack weststack">
+            <div class="busname">Beaverton</div>
+            <div class="bustime west">-- min</div>
+            <div class="bustime smoler west">-- min</div>
+
+        </div>
+        </div>
+        </div>`
     }
 
-    if(placedCircles.length == 1) {
-        circle.y = placedCircles[0].r + circle.r;
-        circle.x = 0;
-        return;
+    updateSize() {
+        this.r = Math.sqrt(.2) * 10;
+
+        this.r = Math.max(...placedCircles.map(circle => circle.r));
+
+        let now = new Date()
+        let timetext = document.querySelector('#hoursCircleText')
+        timetext.innerHTML = 
+        '' + (((now.getHours()+12-1) % 12)+1) + ':' + (now.getMinutes().toString().length == 1 ? '0' : '') + now.getMinutes()
+        if(('' + (((now.getHours()+12-1) % 12)+1)).length == 2) {
+            timetext.classList.add('timeSmallerText')
+        } else {
+            timetext.classList.remove('timeSmallerText')
+        }
+        (async () => {
```

Comment:
I think a function shared between two files might be helpful here, especially because you have a dedicated bus file. Maybe I'm misunderstanding what you're doing here though.

Reply:
Sorry I meant to highlight the whole function

Reply:
Oops- I forgot to delete the bus file (maybe). 
I copy and pasted Micah's code, but I think I should fix it.

Reply:
I didn't see this get addressed in your latest commit, did you somewhere that I didn't see?

Reply:
busBubble.ts no longer exists

Reply:
busbubble.ts may be gone, but the feedback we left on the code there still applies to this function

Reply:
Which was being a duplicate if I'm correct? When I deleted busBubble.ts, I literally just deleted it; it had duplicate functions, but none of the other files were using it, and I did not need to fix anything after removing the file.

Reply:
![image](https://github.com/flamingchickens1540/CLUCK/assets/92497727/94e434d9-5e37-4c35-904f-d0b28e5dbdd0)

Reply:
Oh god! Micah's code is so attrocious I need to put it on r/badcode

Reply:
Not sure if that's the appropriate response...